eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script automates the patch release process for a given library.
+# Note that this assumes that the only changes needed in the patch are upgrades to the dependencies.
+# It makes all the necessary changes and pushes them in an upstream branch which can then be used to trigger the release
+# Please follow the necessary steps for the release - validating all the changes, trigger the release pipeline, approve the release etc.
+# It takes in the following arguments
+# The script takes the following arguments
+# 1. ArtifactName - Name of the library that needs to be patched. This is a required parameter.
+# 2. ServiceDirectoryName - The directory name under 'sdk' folder where the source code of the library is present. This is a required parameter.
+# 3. ReleaseVersion - The latest release version of the library which is to be patched. This is a required parameter.
+# 4. PatchVersion - The version of the patch. This is a required parameter.
+#                   TODO: We can remove the need for this parameter by intepreting this from the first version.
+# 5. BranchName - The name of the remote branch where the changes will be pushed. We push the changes to the 'upstream remote'. This is not a required parameter.
+#                 In case the argument is not provided the branch name is release/{ArtifactName}.
```

Comment:
Since this is to be triggered on a user local box, we will be used the user PAT, which should show the user? Or do you mean the branch name should have the userName?

Reply:
Should we use the name `release/{ArtifactName}/{patchVersion}`? This will allow us to clean up the branch after release.

Reply:
That is a good idea. Done!